woocommerce/woocommerce-ios
# REST API: Fetch site info after authenticating without WPCom

Part of #8453 
 

## Description

This PR integrates the root endpoint to get site info and updates the default site in `DefaultStoresManager` accordingly. Changes include:
- Networking: 
  - Added a new model `WordPressSite` to hold information fetched from `<WP site URL>/wp-json`. This model can be converted to `Site` for use in `DefaultStoresManager`.
  - Add a new remote and mapper to fetch the WP site info.
- Yosemite:
  - Added a new action and store to fetch site WordPress site info given a URL.
  - Updated `AuthenticatedStore` with the new `WordPressStore`.
- UI:
  - Updated `DefaultStoresManager` to load site info with site URL if a placeholder site ID is detected.
 
## Testing instructions

- Enable the feature flag `applicationPasswordAuthenticationForSiteCredentialLogin` and build the app.
- Log out of the app if needed.
- On the prologue screen, select "Enter your site address". If this option is not available, [disable simplified login](https://github.com/woocommerce/woocommerce-ios/blob/8d471f969911067311412d72f58de6ae2c966c6a/WooCommerce/Classes/Authentication/AuthenticationManager.swift#L61) and build the app again.
- Enter your test store address and select "Sign in with site credentials".
- Enter correct credentials and select "Continue". Notice that you are then navigated to the home screen.
- Notice on the home screen that the correct site address and title are displayed.
- Navigate to Menu tab - notice that the correct site title and URL are displayed.

Please feel free to test again with WPCom login to make sure that everything still works as before.

## Screenshots


https://user-images.githubusercontent.com/5533851/209063765-6241cab7-6fd3-4a4a-b1fc-0045df85fdb1.mp4



---
